src/bitmap/store.rs

```diff
@@ -43,6 +43,69 @@ impl Store {
         }
     }
 
+    pub fn insert_range(&mut self, range: Range<u16>) -> u64 {
+        // A Range is defined as being of size 0 if start >= end.
+        if range.is_empty() {
+            return 0;
+        }
+
+        match *self {
+            Array(ref mut vec) => {
+                let (range_start, range_end) = (range.start, range.end);
+
+                // Figure out the starting/ending position in the vec
+                let pos_start = unwrap_either(vec.binary_search(&range_start));
```

Comment:
You also have used the `Result::unwrap_or_else` method, too! Not sure this is better :)

```suggestion
                let pos_start = vec.binary_search(&range_start).unwrap_or_else(|x| x);
```

Reply:
Ah neat, I didn't realise `unwrap_or_else()` passed in the err! Like it 👍